diff -U0 emacs-23.1/src/ChangeLog.last-tty emacs-23.1/src/ChangeLog
--- emacs-23.1/src/ChangeLog.last-tty	2009-07-29 18:55:12.000000000 +0200
+++ emacs-23.1/src/ChangeLog	2011-10-13 16:14:25.642627196 +0200
@@ -0,0 +1,4 @@
+2009-11-25  Jan Djärv  <jan.h.d@swipnet.se>
+
+	* term.c (delete_tty): Remove check for last terminal (bug#4970).
+
diff -up emacs-23.1/src/term.c.last-tty emacs-23.1/src/term.c
--- emacs-23.1/src/term.c.last-tty	2009-06-29 05:34:26.000000000 +0200
+++ emacs-23.1/src/term.c	2011-10-13 16:13:35.568411522 +0200
@@ -3959,8 +3959,6 @@ static void
 delete_tty (struct terminal *terminal)
 {
   struct tty_display_info *tty;
-  Lisp_Object tail, frame;
-  int last_terminal;
 
   /* Protect against recursive calls.  delete_frame in
      delete_terminal calls us back when it deletes our last frame.  */
@@ -3972,19 +3970,6 @@ delete_tty (struct terminal *terminal)
 
   tty = terminal->display_info.tty;
 
-  last_terminal = 1;
-  FOR_EACH_FRAME (tail, frame)
-    {
-      struct frame *f = XFRAME (frame);
-      if (FRAME_LIVE_P (f) && (!FRAME_TERMCAP_P (f) || FRAME_TTY (f) != tty))
-        {
-          last_terminal = 0;
-          break;
-        }
-    }
-  if (last_terminal)
-      error ("Attempt to delete the sole terminal device with live frames");
-
   if (tty == tty_list)
     tty_list = tty->next;
   else
