diff --git a/src/xfaces.c b/src/xfaces.c
index 9eb3d36..87340af 100644
--- a/src/xfaces.c
+++ b/src/xfaces.c
@@ -5776,6 +5776,7 @@ realize_face (cache, attrs, former_face_id)
       struct face *former_face = cache->faces_by_id[former_face_id];
       uncache_face (cache, former_face);
       free_realized_face (cache->f, former_face);
+      SET_FRAME_GARBAGED (cache->f);
     }
 
   if (FRAME_WINDOW_P (cache->f))
