diff -up emacs-23.1/src/font.c.font emacs-23.1/src/font.c
--- emacs-23.1/src/font.c.font	2009-07-28 16:51:20.000000000 +0200
+++ emacs-23.1/src/font.c	2010-04-04 00:26:12.000000000 +0200
@@ -1,6 +1,6 @@
 /* font.c -- "Font" primitives.
-   Copyright (C) 2006, 2007, 2008, 2009 Free Software Foundation, Inc.
-   Copyright (C) 2006, 2007, 2008, 2009
+   Copyright (C) 2006, 2007, 2008, 2009, 2010 Free Software Foundation, Inc.
+   Copyright (C) 2006, 2007, 2008, 2009, 2010
      National Institute of Advanced Industrial Science and Technology (AIST)
      Registration Number H13PRO009
 
@@ -23,6 +23,7 @@ along with GNU Emacs.  If not, see <http
 #include <stdio.h>
 #include <stdlib.h>
 #include <ctype.h>
+#include <setjmp.h>
 
 #include "lisp.h"
 #include "buffer.h"
@@ -1562,6 +1576,18 @@ font_parse_fcname (name, font)
 	      prop = font_intern_prop ("light", 5, 1);
 	      FONT_SET_STYLE (font, FONT_WEIGHT_INDEX, prop);
 	    }
+	  else if (PROP_MATCH ("Book", 4))
+	    {
+	      prop_found = 1;
+	      prop = font_intern_prop ("book", 4, 1);
+	      FONT_SET_STYLE (font, FONT_WEIGHT_INDEX, prop);
+	    }
+	  else if (PROP_MATCH ("Medium", 6))
+	    {
+	      prop_found = 1;
+	      prop = font_intern_prop ("medium", 6, 1);
+	      FONT_SET_STYLE (font, FONT_WEIGHT_INDEX, prop);
+	    }
 	  else if (PROP_MATCH ("Semi-Bold", 9))
 	    {
 	      prop_found = 1;
@@ -1586,6 +1612,18 @@ font_parse_fcname (name, font)
 	      prop = font_intern_prop ("oblique", 7, 1);
 	      FONT_SET_STYLE (font, FONT_SLANT_INDEX, prop);
 	    }
+	  else if (PROP_MATCH ("Semi-Condensed", 14))
+	    {
+	      prop_found = 1;
+	      prop = font_intern_prop ("semi-condensed", 14, 1);
+	      FONT_SET_STYLE (font, FONT_WIDTH_INDEX, prop);
+	    }
+	  else if (PROP_MATCH ("Condensed", 9))
+	    {
+	      prop_found = 1;
+	      prop = font_intern_prop ("condensed", 9, 1);
+	      FONT_SET_STYLE (font, FONT_WIDTH_INDEX, prop);
+	    }
 	  else {
 	    if (prop_found)
 	      return -1; /* Unknown property in GTK-style font name.  */
